diff --git a/dist/lib/better-sqlite3.js b/dist/lib/better-sqlite3.js
index 3cbc6a100647050c936fedb22e3f26168b82ad99..1846654392ef5ebe0c1942f617d665ef9ebbadcf 100644
--- a/dist/lib/better-sqlite3.js
+++ b/dist/lib/better-sqlite3.js
@@ -14,7 +14,7 @@ class BetterSqlite3Instrumentation extends instrumentation_1.InstrumentationBase
         return [
             new instrumentation_1.InstrumentationNodeModuleDefinition('better-sqlite3', supportedVersions, (moduleExports, moduleVersion) => {
                 api_1.diag.debug(`Applying patch for better-sqlite3@${moduleVersion}`);
-                const proto = moduleExports.prototype;
+                const proto = moduleExports.prototype || moduleExports.default.prototype;
                 if (!(0, instrumentation_1.isWrapped)(proto.exec)) {
                     this._wrap(proto, 'exec', this.patchExec);
                     this._wrap(proto, 'prepare', this.patchPrepare);
@@ -23,8 +23,9 @@ class BetterSqlite3Instrumentation extends instrumentation_1.InstrumentationBase
                 return moduleExports;
             }, (moduleExports, moduleVersion) => {
                 if (moduleExports !== undefined) {
+                    const proto = moduleExports.prototype || moduleExports.default.prototype;
                     api_1.diag.debug(`Removing patch for better-sqlite3@${moduleVersion}`);
-                    this._massUnwrap([moduleExports.prototype], ['exec', 'prepare', 'pragma']);
+                    this._massUnwrap([proto], ['exec', 'prepare', 'pragma']);
                 }
             }),
         ];
